presto-native-execution/presto_cpp/main/PrestoTask.cpp

```diff
@@ -236,7 +239,25 @@ protocol::TaskInfo PrestoTask::updateInfoLocked() {
 
   // TODO(venkatra): Populate these memory stats as well.
   prestoTaskStats.revocableMemoryReservationInBytes = {};
-  prestoTaskStats.cumulativeUserMemory = {};
+
+  // Set the lastTaskStatCallMs to execution start time if it is 0.
+  if (lastTaskStatCallMs == 0) {
+    lastTaskStatCallMs = taskStats.executionStartTimeMs;
+  }
+
+  uint16_t currentTimeMs = velox::getCurrentTimeMs();
+  uint16_t sinceLastPeriodMillis = currentTimeMs - lastTaskStatCallMs;
+
+  int64_t currentUserBytes = tracker->currentBytes();
+
+  int64_t averageUserMemoryForLastPeriod =
+      (currentUserBytes + lastUserMemoryReservation) / 2;
+
+  prestoTaskStats.cumulativeUserMemory +=
+      (averageUserMemoryForLastPeriod * sinceLastPeriodMillis) / 1000;
+
+  lastTaskStatCallMs = currentTimeMs;
```

Comment:
Can you add a test in presto-native-execution/presto_cpp/main/tests/TaskManagerTest.cpp? Thanks!